src/lib.rs

```diff
@@ -38,19 +38,25 @@ impl Cloudinary {
         src: String,
         options: &UploadOptions<'_>,
     ) -> Result<UploadResult> {
-        let client = Client::new();
         let file = prepare_file(&src).await?;
         let multipart = self.build_form_data(options).part("file", file);
-        let response = client
-            .post(format!(
-                "https://api.cloudinary.com/v1_1/{}/image/upload",
-                self.cloud_name
-            ))
-            .multipart(multipart)
-            .send()
-            .await?;
-        let text = response.text().await?;
-        let json = serde_json::from_str(&text).context(format!("failed to parse:\n\n {}", text))?;
+        let json = self.make_request(multipart).await?;
+        Ok(json)
+    }
+
+    /**
+     Handles image upload from url
+     
+      ```rust
+      use cloudinary::{Cloudinary, upload::UploadOptions};
+      let options = UploadOptions::new().set_public_id("gojo.jpg".to_string());
+      let cloudinary = Cloudinary::new("api_key".to_string(), "cloud_name".to_string(), "api_secret".to_string());
+      let result = cloudinary.upload_image_from_url("https://some.com/assets/satoru_gojo.jpg".to_string(), &options);
+      ```
+     */
+    pub async fn upload_image_from_url(&self, src: String, options: &UploadOptions<'_>) -> Result<UploadResult> {
```

Comment:
Let's expose a single upload method to the users. We can check whether that is a local file or not and construct the form accordingly.